### PR TITLE
Support deserializing floats without fractional component.

### PR DESCRIPTION
**Pull request by FSMaxB**
This is the implementation of #24.

### CODE REVIEW

File: .github/workflows/msrv.yml [hunk 1]
```diff
@@ -26,4 +26,4 @@ jobs:
       uses: actions-rs/cargo@v1
       with:
         command: check
-        args: --features std,serde,lax_deserialize
+        args: --all-features
```
Comment (FSMaxB) [inline .github/workflows/msrv.yml:29]: Not sure if you want to run the same suite of tests on the MSRV or not.

File: src/int.rs [hunk 2]
```diff
@@ -594,6 +599,17 @@ impl<'de> Deserialize<'de> for Int {
     }
 }
 
+#[cfg(feature = "float_deserialize")]
+#[inline(always)]
+pub(crate) fn is_acceptable_float(float: f64) -> bool {
```
Comment (FSMaxB) [inline src/int.rs:604]: I could put this into a separate module but I'm not sure what to call that. Or I could put this in `lib.rs`.

File: tests/uint.rs [hunk 3]
```diff
@@ -18,38 +19,50 @@ fn assert_serialize(number: u32) {
...
-    assert_eq!(deserialize_uint_from(9007199254740991.0).unwrap(), UInt::MAX);
-    assert_eq!(deserialize_uint_from(9007199254740991.49).unwrap(), UInt::MAX);
-    assert!(deserialize_uint_from(9007199254740992.0).is_err());
+#[cfg_attr(not(feature = "float_deserialize"), ignore)]
+fn deserialize_integral_float() {
+    assert_eq!(deserialize_from(1.0).unwrap(), uint!(1));
+    assert_eq!(deserialize_from(9007199254740991.0).unwrap(), UInt::MAX);
+    assert!(deserialize_from(9007199254740992.0).is_err());
+    // NOTE: This still ends up as integral because the .49 exceeds the representable range of f64
+    assert_eq!(deserialize_from(9007199254740991.49).unwrap(), UInt::try_from(9007199254740991i64).unwrap());
```
Comment (FSMaxB) [inline tests/uint.rs:51]: This is unavoidable unfortunately.

File: tests/uint.rs [hunk 4]
```diff
@@ -18,38 +19,50 @@ fn assert_serialize(number: u32) {
...
 #[cfg_attr(feature = "lax_deserialize", ignore)]
-fn strict_deserialize_uint() {
-    assert!(deserialize_uint_from(0.5).is_err());
-    assert!(deserialize_uint_from(1.0).is_err());
-    assert!(deserialize_uint_from(9007199254740991.0).is_err());
-    assert!(deserialize_uint_from(9007199254740991.49).is_err());
-    assert!(deserialize_uint_from(9007199254740992.0).is_err());
+fn dont_deserialize_fractional_float() {
+    assert!(deserialize_from(0.5).is_err());
+    assert!(deserialize_from(42.1337).is_err());
```
Comment (FSMaxB) [inline tests/uint.rs:41]: I could also add something like 2^52 - 0.5, that would be a fractional double that uses it's entire mantissa if I am not mistaken.

**Comment by FSMaxB** [comment]
I'm still unsure about the documentation.

File: .github/workflows/msrv.yml (hunk 1 above)
Comment (jplatte) [reply .github/workflows/msrv.yml:29]: Yeah. There used to be a feature that only worked on Nightly, but that has been removed.

File: src/int.rs (hunk 2 above)
Comment (jplatte) [reply src/int.rs:602]: Moving it to `lib.rs` sounds good.

File: tests/uint.rs (hunk 4 above)
Comment (jplatte) [reply tests/uint.rs:41]: Feel free to add whatever tests you feel are worthwhile.

File: README.md [hunk 5]
```diff
@@ -34,10 +34,12 @@ implementations of `std::error::Error` for `ParseIntError` and
 without the `std` feature.
 
 Deserialization can be routed through `f64` instead of `u64` with the
-`lax_deserialize` feature. If a deserialized number is fractional, the
-fractional part is discarded. Please be aware that `serde_json` doesn't
-losslessly parse large floats with a fractional part by default (even if the
-fractional part is `.0`). To fix that, enable its `float_roundtrip` feature.
+`float_deserialize` feature. This will still not deserialize numbers with a
+fractional component. Enabling the `lax_deserialize` feature will discard the
```
Comment (jplatte) [inline README.md:38]:
```suggestion
non-zero fractional component. Enabling the `lax_deserialize` feature will discard the
```

Comment (jplatte) [reply README.md:38]: Otherwise docs look good to me.

Comment (jplatte) [reply README.md:38]: Maybe "still not deserialize" could be changed to "still reject"

File: tests/uint.rs (hunk 4 above)
Comment (FSMaxB) [reply tests/uint.rs:41]: Then I'm going to leave this out since I don't think there's anything fundamentally different from the existing numbers.

**Pull request merged**